src/treebuilders/DerivativeCalculator.cpp

```diff
@@ -261,18 +261,19 @@ template <int D> void DerivativeCalculator<D>::applyOperator(OperatorState<D> &o
 
 /** Perform the required linear algebra operations in order to apply an
 operator component to a f-node in a n-dimensional tensor space. */
-template <int D> void DerivativeCalculator<D>::tensorApplyOperComp(OperatorState<D> &os) {
-    double **aux = os.getAuxData();
+template <int D, typename T> void DerivativeCalculator<D, T>::tensorApplyOperComp(OperatorState<D, T> &os) {
+    T **aux = os.getAuxData();
     double **oData = os.getOperData();
+    /*
```

Comment:
Should the commented-out code be removed altogether? Eigen does have the possibility of offloading to BLAS/LAPACK https://eigen.tuxfamily.org/dox/TopicUsingBlasLapack.html